src/frontend/sharedComponents/PermissionAudio.tsx

```diff
@@ -1,72 +1,100 @@
-};
-
-const handleOpenSettings = () => {
-  Linking.openSettings();
-};
+import {BottomSheetModalMethods} from '@gorhom/bottom-sheet/lib/typescript/types';
+import {PermissionResponse} from 'expo-modules-core';
+import {useNavigationFromRoot} from '../hooks/useNavigationWithTypes';
 
-interface PermissionAudio {
-  sheetRef: React.RefObject<BottomSheetModalMethods>;
+const m = defineMessages({
+  title: {
+    id: 'screens.AudioPermission.title',
+    defaultMessage: 'Recording Audio with CoMapeo',
+    description: 'Screen title for audio permission screen',
+  },
+  description: {
+    id: 'screens.AudioPermission.description',
+    defaultMessage:
+      'To record audio while using the app and in the background CoMapeo needs to access your microphone. Please enable microphone permissions in your app settings.',
+    description: 'Screen description for audio permission screen',
+  },
+  notNowButtonText: {
+    id: 'screens.AudioPermission.Button.notNow',
+    defaultMessage: 'Not Now',
+    description: 'Screen button text for not allowed audio permission',
+  },
+  allowButtonText: {
+    id: 'screens.AudioPermission.Button.allow',
+    defaultMessage: 'Allow',
+    description: 'Screen button text for allow the audio permission',
+  },
+  goToSettingsButtonText: {
+    id: 'screens.AudioPermission.Button.goToSettings',
+    defaultMessage: 'Go to Settings',
+    description:
+      'Screen button text for navigate user to settings when audio permission was denied',
+  },
+});
+interface PermissionAudioProps {
   closeSheet: () => void;
   isOpen: boolean;
+  sheetRef: React.RefObject<BottomSheetModalMethods>;
 }
 
-export const PermissionAudio: FC<PermissionAudio> = props => {
-  const {sheetRef, closeSheet, isOpen} = props;
+export const PermissionAudio: FC<PermissionAudioProps> = ({
+  closeSheet,
+  isOpen,
+  sheetRef,
+}) => {
   const {formatMessage: t} = useIntl();
-  const [permissionResponse] = Audio.usePermissions({request: false});
+  const navigation = useNavigationFromRoot();
+  const [permissionResponse, setPermissionResponse] =
+    useState<PermissionResponse | null>(null);
 
-  const handlePermissionGranted = useCallback(() => {
+  const handleOpenSettings = () => {
+    Linking.openSettings();
     closeSheet();
-  }, [closeSheet]);
-
-  const isPermissionGranted = Boolean(permissionResponse?.granted);
+  };
 
-  useEffect(() => {
-    if (isPermissionGranted) handlePermissionGranted();
-  }, [isPermissionGranted, handlePermissionGranted]);
+  const handleRequestPermission = async () => {
+    const response = await Audio.requestPermissionsAsync();
+    setPermissionResponse(response);
+    if (response.status === 'granted') {
+      closeSheet();
+      navigation.navigate('Audio');
+    } else if (response.status === 'denied' && response.canAskAgain) {
+      closeSheet();
+    } else if (response.status === 'denied' && !response.canAskAgain) {
+      handleOpenSettings();
+    }
```

Comment:
is this logic backwards?

Reply:
It actually isn't. I am not sure of the logic behind why "canAskAgain" is what it is, I just observed how it worked. If your response is canAskAgain, it is immediately after your very first response to the device query asking for Permission -- the first time you answer it saying, no, I don't give permission, and so immediately after that, in that case you don't need to go to settings and can just have the sheet close. 

If it's a response of !canAskAgain, you have already said you don't give permission but are asking for using audio again, so now you need to go to settings to understand what it is we are asking you (which is I assume what codesign wanted).

I am not sure if I explained that well! I can walk you through it over screen share if that's easier.